charts/hivemq-platform/templates/hivemq-custom-resource.yml

```
@@ -7,11 +7,18 @@ apiVersion: hivemq.com/v1
 kind: HiveMQPlatform
 metadata:
   name: "{{ .Release.Name }}"
+  {{- if and (typeIsLike "map[string]interface {}" .Values.platform.annotations) (.Values.platform.annotations) }}
+  annotations:
+  {{- .Values.platform.annotations | toYaml | nindent 4 }}
+  {{- end }}
```

checking the type is not really needed, as this validation is done by the JSON schema. Also, you could just simplify this expression by using a `with` statement (this is just a personal preference, `if` is also fine).

```suggestion
  {{- with .Values.platform.annotations }}
  annotations:
  {{- toYaml . | nindent 4 }}
  {{- end }}
```